src/main/java/com/csemosip/bookingservice/service/impl/BookingServiceImpl.java

```diff
@@ -36,7 +40,27 @@ public List<Booking> findAllBookings() {
     @Override
     public Booking createBooking(BookingDTO bookingDTO) {
         Resource resource = resourceRepository.findById(bookingDTO.getResourceId())
-                .orElseThrow(() -> new ResourceNotFoundException("Resource not found with ID: " + bookingDTO.getResourceId()));
+                .orElseThrow(() -> new ResourceNotFoundException("Resource not found with id: " + bookingDTO.getResourceId()));
+
+        // check if the booking count exceeds the max count of the resource
+        if (bookingDTO.getCount() > resource.getCount())
+            throw new ValidationErrorException("count cannot exceed the max count of the resource.");
+
+        // check if the booking count exceeds the MIN available count during the timeslot
+        List<ResourceAvailabilityDTO> availabilityList = resourceService.getAvailabilityByResourceIdAndTimeslot(
+                resource.getId(),
+                bookingDTO.getStartTime(),
+                bookingDTO.getEndTime()
+        );
+
+        int minAvailable = resource.getCount();
+        for (ResourceAvailabilityDTO availabilityDTO : availabilityList) {
+            if (availabilityDTO.getCount() < minAvailable)
+                minAvailable = availabilityDTO.getCount();
```

Comment:
Instead of finding the minimum available count overall first and then comparing with the count in the booking, we can just compare with the count in the booking here itself and throw the exception. (`if (availabilityDTO.getCount() < bookingDTO.getCount())`)
That way, if the first slot itself doesn't have enough resources, then we can throw the exception earlier.

Reply:
True. I will change the code.